src/main/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDIProvider.scala

```diff
@@ -325,10 +328,20 @@ private[impl] object DVDIProviderValidations extends ExternalVolumeValidations {
   private[this] def matchesProvider(volume: ExternalVolume): Boolean = volume.external.provider == name
   private[this] def matchesProviderRaml(volume: AppExternalVolume): Boolean = volume.external.provider.contains(name)
 
+  private[this] def isForUniquenessCheck(volume: ExternalVolume): Boolean = !volume.external.shared
+  private[this] def isForUniquenessCheckRaml(volume: AppExternalVolume): Boolean = !volume.external.shared
+
   private[this] def namesOfMatchingVolumes(app: AppDefinition): Seq[String] =
-    app.externalVolumes.withFilter(matchesProvider).map(_.external.name)
+    app
+      .externalVolumes
+      .withFilter(matchesProvider)
+      .withFilter(isForUniquenessCheck)
```

Comment:
Hm, this is a little … problematic: If one app declares volume `foo` as shared, and another doesn't (assuming it's not shared), the one with the shared declaration will be ignored, and the validation for the other one will pass.

I don't think this is a general issue, since users should only turn this on if their volumes can be shared. However, We should point this out in the docs/release notes: having 3 references to the same volume where only one is **not shared** is totally fine and won't be rejected.

Reply:
Yeah, that is a bit ugly. The correct thing would be to require that *all* volumes with the same name have `shared=true`. I'll look into it

Reply:
Ok, this is actually not a problem, the validation behaves as I described it:

The code here is only to gather a list of all existing volume names, but each volume - even the ones with `shared=true` are checked against this list, so if one volume without the flag exists, it prevents the volumes with `shared=true` to be validated.

Reply:
I've added another unit test to ensure this behaviour, and added a comment to the changelog